functions/coerce.js

```diff
@@ -19,34 +19,42 @@ const coerce = (version, options) => {
 
   let match = null
   if (!options.rtl) {
-    match = version.match(re[t.COERCE])
+    match = version.match(options.includePrerelease ? re[t.COERCEFULL] : re[t.COERCE])
   } else {
     // Find the right-most coercible string that does not share
     // a terminus with a more left-ward coercible string.
     // Eg, '1.2.3.4' wants to coerce '2.3.4', not '3.4' or '4'
+    // With includePrerelease option set, '1.2.3.4-pre.1.2' wants to coerce '2.3.4-pre.1.2', not '1.2'
     //
     // Walk through the string checking with a /g regexp
-    // Manually set the index so as to pick up overlapping matches.
+    // Manually set the index so, as to pick up overlapping matches.
```

Comment:
"[so as to](https://dictionary.cambridge.org/dictionary/english/so-as-to)" is a valid phrase, this comma is not warranted.